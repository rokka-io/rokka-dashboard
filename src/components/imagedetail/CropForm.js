import PropTypes from 'prop-types'
import React from 'react'

const CropForm = ({
  onChange,
  x = '',
  y = '',
  width = '',
  height = '',
  disableWidthHeight = false
}) => (
  <div className="row">
    <div className="col-md-3 col-sm-3">
      <div className="rka-form-group">
        <label className="rka-label">Offset X</label>
        <input type="text" name="x" className="rka-input-txt" value={x} onChange={onChange} />
      </div>
    </div>
    <div className="col-md-3 col-sm-3">
      <div className="rka-form-group">
        <label className="rka-label">Offset Y</label>
        <input type="text" name="y" className="rka-input-txt" value={y} onChange={onChange} />
      </div>
    </div>
    <div className="col-md-3 col-sm-3">
      <div className="rka-form-group">
        <label className="rka-label">Width</label>
        <input
          type="text"
          name="width"
          disabled={disableWidthHeight}
          className="rka-input-txt"
          value={width}
          onChange={onChange}
        />
      </div>
    </div>
    <div className="col-md-3 col-sm-3">
      <div className="rka-form-group">
        <label className="rka-label">Height</label>
        <input
          type="text"
          name="height"
          disabled={disableWidthHeight}
          className="rka-input-txt"
          value={height}
          onChange={onChange}
        />
      </div>
    </div>
  </div>
)
CropForm.propTypes = {
  onChange: PropTypes.func.isRequired,
  x: PropTypes.number,
  y: PropTypes.number,
  width: PropTypes.number,
  height: PropTypes.number,
  disableWidthHeight: PropTypes.bool
}

export default CropForm
