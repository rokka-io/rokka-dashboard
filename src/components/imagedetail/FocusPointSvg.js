import PropTypes from 'prop-types'
import React, { forwardRef } from 'react'
import cx from 'classnames'
import focusPointIcon from '../../img/focuspoint-icon.svg'

const FocusPointSvg = forwardRef(({ x, y, isVisible = false }, ref) => (
  <div className={cx('rka-focuspoint-icon-container', {'vis-h': !isVisible})} style={{top: y, left: x}} ref={ref}>
    <svg className="rka-focuspoint-icon">
      <use xlinkHref={focusPointIcon + '#focuspointicon'} />
    </svg>
  </div>
))
FocusPointSvg.propTypes = {
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  isVisible: PropTypes.bool
}

export default FocusPointSvg
