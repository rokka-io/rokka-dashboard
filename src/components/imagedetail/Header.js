import PropTypes from 'prop-types'
import React from 'react'
import cx from 'classnames'
import moment from 'moment'
import focusIcon from '../../img/focus-icon.svg'
import linkicon from '../../img/link-icon.svg'

const Header = ({ image, focusMenuActive, onClickToggleFocusMenu, onClickSave }) => (
  <div className="rka-content-header clearfix">
    <div className="flo-l">
      <h1 className="rka-h1">{image.name}</h1>
      <p className="txt-gray-darker mt-sm txt-sm txt-italic">
        <span className="mr-sm">{moment(image.created).fromNow()},</span>
        <span className="mr-sm">{`${Math.round(image.size / 1024)}KB`},</span>
        <span className="mr-sm">{[image.width || '?', image.height || '?'].join('×')},</span>
        <span className="mr-sm">{image.format.toUpperCase()},</span>
        <span className="">
          <a
            className="rka-link rka-hash-link"
            href={image.url}
            target="_blank"
            rel="noopener noreferrer"
          >
            {image.hash}
            <svg className="rka-link-icon">
              <use xlinkHref={linkicon + '#linkicon'} />
            </svg>
          </a>
        </span>
        <span style={{ marginLeft: 30 }}>
          <a
            className="rka-link rka-hash-link"
            href={image.url.replace(image.hash, image.short_hash)}
            target="_blank"
            rel="noopener noreferrer"
          >
            {image.short_hash}
            <svg className="rka-link-icon">
              <use xlinkHref={linkicon + '#linkicon'} />
            </svg>
          </a>
        </span>
      </p>
    </div>
    <div className="flo-r">
      <button
        className={cx('rka-focus-trigger', 'mr-md', { 'is-active': focusMenuActive })}
        onClick={onClickToggleFocusMenu}
      >
        {focusMenuActive ? (
          <span className="rka-focus-icon txt-c txt-lg">×</span>
        ) : (
          <svg className="rka-focus-icon">
            <use xlinkHref={focusIcon + '#focusicon'} />
          </svg>
        )}
      </button>
      <button className="rka-button rka-button-brand mt-sm" onClick={onClickSave}>
        Save
      </button>
    </div>
  </div>
)
Header.propTypes = {
  image: PropTypes.shape({
    name: PropTypes.string.isRequired,
    created: PropTypes.string.isRequired,
    size: PropTypes.number.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    format: PropTypes.string.isRequired,
    hash: PropTypes.string.isRequired
  }).isRequired,
  focusMenuActive: PropTypes.bool.isRequired,
  onClickToggleFocusMenu: PropTypes.func.isRequired,
  onClickSave: PropTypes.func.isRequired
}

export default Header
