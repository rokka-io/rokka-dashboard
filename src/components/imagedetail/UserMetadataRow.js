import PropTypes from 'prop-types'
import React from 'react'

const ROKKA_TYPES = [
  {label: 'String', value: 'str'},
  {label: 'Int', value: 'int'},
  {label: 'Double', value: 'double'},
  {label: 'Date', value: 'date'},
  {label: 'Location', value: 'latlon'},
  {label: 'Array', value: 'array'}
]

const UserMetadataRow = ({ index, onChange, onClickRemove, onClickAdd, isNew = false, name = '', type = 'str', value = '' }) => (
  <div key={index} className="rka-image-metadata-container mb-xs">
    {!isNew
      ? <button className="rka-close-icon" onClick={(e) => onClickRemove(e, index)} />
      : null
    }
    <div className="row">
      <div className="col-md-4">
        <div className="rka-form-group mb-n">
          <input
            type="text"
            value={name}
            className="rka-input-txt"
            placeholder="Name"
            onChange={(e) => onChange(isNew, index, {name: e.currentTarget.value})}
          />
        </div>
      </div>
      <div className="col-md-4">
        <div className="rka-form-group mb-n">
          <select className="rka-select" value={type} onChange={(e) => onChange(isNew, index, {type: e.currentTarget.value})}>
            {ROKKA_TYPES.map(option => (
              <option key={option.value} value={option.value}>{option.label}</option>
            ))}
          </select>
        </div>
      </div>
      <div className="col-md-4">
        <div className="rka-form-group mb-n">
          <input
            type="text"
            value={value}
            className="rka-input-txt"
            placeholder="Value"
            onChange={(e) => onChange(isNew, index, {value: e.currentTarget.value})}
          />
        </div>
      </div>
    </div>
    {isNew
      ? <button type="button" className="rka-button rka-button-brand rka-button-sm mt-md" onClick={onClickAdd}>Add metadata</button>
      : null
    }
  </div>
)
UserMetadataRow.propTypes = {
  index: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  onClickRemove: PropTypes.func.isRequired,
  onClickAdd: PropTypes.func.isRequired,
  isNew: PropTypes.bool,
  name: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)])
}

export default UserMetadataRow
