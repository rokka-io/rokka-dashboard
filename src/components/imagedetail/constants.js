export const FOCUS_POINT = 'focusPoint'
export const FOCUS_AREA = 'focusArea'
