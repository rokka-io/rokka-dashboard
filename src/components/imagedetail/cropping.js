/**
 * Image cropping functions
 */

/**
 * Get the offset of an element
 *
 * From react-image-crop
 *
 * @param {Element} el
 * @return {{top: number, left: number}}
 */
function getElementOffset(el) {
  const rect = el.getBoundingClientRect()
  const docEl = document.documentElement

  const rectTop = rect.top + window.pageYOffset - docEl.clientTop
  const rectLeft = rect.left + window.pageXOffset - docEl.clientLeft

  return {
    top: rectTop,
    left: rectLeft
  }
}

/**
 * Get the mouse position of a click/touch.
 *
 * From react-image-crop
 *
 * @param {MouseEvent} e
 * @return {{x: number, y: number}}
 */
function getClientPos(e) {
  if (e.touches) {
    return { x: e.touches[0].pageX, y: e.touches[0].pageY }
  }
  return { x: e.pageX, y: e.pageY }
}

/**
 * Calculate the rendered scale of an image
 *
 * @param {number} naturalWidth
 * @param {number} width
 * @return {number}
 */
export function calculateScale(naturalWidth, width) {
  return naturalWidth / width
}

/**
 * Calculates the click position within the parent element of the clicked element.
 *
 * @param {MouseEvent} evt
 * @return {{x: number, y: number}}
 */
export function calculateClickPosition(evt) {
  const clientPos = getClientPos(evt)
  const parentElOffset = getElementOffset(evt.currentTarget.parentElement)

  return {
    x: Math.round(clientPos.x - parentElOffset.left),
    y: Math.round(clientPos.y - parentElOffset.top)
  }
}

/**
 * Apply scale to an x and y position.
 *
 * @param {{x: number, y: number}} pos
 * @param {number} scale
 * @return {{x: number, y: number}}
 */
export function scalePosition({ x, y }, scale) {
  return {
    x: Math.round(x * scale),
    y: Math.round(y * scale)
  }
}

/**
 * Calculates the position for rendering a target element.
 *
 * @param {number} scale
 * @param {number} targetElementWidth
 * @param {{x: number, y: number}} pos
 * @return {{x: number, y: number}}
 */
export function calculateRenderedPosition(scale, targetElementWidth, pos) {
  const focusPointSize = (targetElementWidth / 2) * scale
  const x = pos.x - focusPointSize
  const y = pos.y - focusPointSize

  return scalePosition({ x, y }, 1 / scale)
}

/**
 * Converts a crop area from pixel values to relative percent pixel values based on the width/height of the
 * image.
 *
 * @param {{x: number, y: number, width: number, height: number}} crop
 * @param {number} width  Real image width (naturalWidth)
 * @param {number} height Real image height (naturalHeight)
 * @returns {{x: number, y: number, width: number, height: number}}
 */
export function pixelToPercent(crop, width, height) {
  return {
    x: (crop.x * 100) / width,
    y: (crop.y * 100) / height,
    width: (crop.width * 100) / width,
    height: (crop.height * 100) / height
  }
}
