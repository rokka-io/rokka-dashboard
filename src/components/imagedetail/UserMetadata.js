import React, { PropTypes } from 'react'
import UserMetadataRow from './UserMetadataRow'

const UserMetadata = ({ metadata, addMetadata, onChange, onClickRemove, onClickAdd }) => (
  <section className="rka-box mb-n">
    <div className="row">
      <div className="col-md-7">
        <h2 className="rka-h2 mb-md">User metadata</h2>
        {metadata.map((data, index) => (
          <UserMetadataRow key={index} index={index} onChange={onChange} onClickRemove={onClickRemove} onClickAdd={onClickAdd} {...data} />
        ))}
        <UserMetadataRow isNew index={metadata.length} onChange={onChange} onClickRemove={onClickRemove} onClickAdd={onClickAdd} {...addMetadata} />
      </div>
    </div>
  </section>
)
UserMetadata.propTypes = {
  metadata: PropTypes.array.isRequired,
  addMetadata: PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string
  }).isRequired,
  onChange: PropTypes.func.isRequired,
  onClickRemove: PropTypes.func.isRequired,
  onClickAdd: PropTypes.func.isRequired
}

export default UserMetadata
