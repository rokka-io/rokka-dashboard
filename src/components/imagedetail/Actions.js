import React, { PropTypes } from 'react'
import cx from 'classnames'
import { FOCUS_POINT, FOCUS_AREA } from './constants'
import CropForm from './CropForm'
import focusPointIcon from '../../img/focuspoint-icon.svg'
import focusAreaIcon from '../../img/focusarea-icon.svg'

const Actions = ({ menuActive, actionsActive, onChange, onToggleActions, focusArea = {}, focusType = null }) => (
  <div className={cx('rka-crop-actions', {'is-active': menuActive})}>
    <div className={cx('rka-crop-action-container', {'is-active': actionsActive})}>
      <a href="#" id="focusPoint" onClick={onToggleActions} className={cx('rka-crop-action', { 'is-active': focusType === FOCUS_POINT })}>
        <svg className="rka-focuspoint-icon">
          <use xlinkHref={focusPointIcon + '#focuspointicon'} />
        </svg>
      </a>
      <a href="#" id="focusArea" onClick={onToggleActions} className={cx('rka-crop-action', { 'is-active': focusType === FOCUS_AREA })}>
        <svg className="rka-focusarea-icon">
          <use xlinkHref={focusAreaIcon + '#focusareaicon'} />
        </svg>
      </a>
    </div>
    <div className={cx('rka-crop-actions-form', {'is-active': actionsActive})}>
      { actionsActive
        ? <CropForm onChange={onChange} disableWidthHeight={focusType === FOCUS_POINT} {...focusArea} />
        : null
      }
    </div>
    { actionsActive
      ? <div className="rka-crop-actions-close-container">
        <a href="#" onClick={onToggleActions} className="rka-crop-actions-close">Ã—</a>
      </div>
      : null
    }
  </div>
)
Actions.propTypes = {
  menuActive: PropTypes.bool.isRequired,
  actionsActive: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  onToggleActions: PropTypes.func.isRequired,
  focusArea: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number,
    width: PropTypes.number,
    height: PropTypes.number
  }),
  focusType: PropTypes.string
}

export default Actions
