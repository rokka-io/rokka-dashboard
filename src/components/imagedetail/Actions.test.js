import React from 'react'
import Actions from '../../../src/components/imagedetail/Actions'
import { FOCUS_POINT, FOCUS_AREA } from '../../../src/components/imagedetail/constants'
import renderer from 'react-test-renderer'

test('Actions does render', () => {
  const component = renderer.create(
    <Actions menuActive actionsActive onChange={() => {}} onToggleActions={() => {}} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Actions does render with menu inactive', () => {
  const component = renderer.create(
    <Actions menuActive={false} actionsActive onChange={() => {}} onToggleActions={() => {}} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Actions does render with menu/actions inactive', () => {
  const component = renderer.create(
    <Actions
      menuActive={false}
      actionsActive={false}
      onChange={() => {}}
      onToggleActions={() => {}}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Actions does render with focus area', () => {
  const component = renderer.create(
    <Actions
      menuActive
      actionsActive
      onChange={() => {}}
      onToggleActions={() => {}}
      focusType={FOCUS_AREA}
      focusArea={{
        x: 1,
        y: 1,
        width: 100,
        height: 100
      }}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Actions does render with focus point', () => {
  const component = renderer.create(
    <Actions
      menuActive
      actionsActive
      onChange={() => {}}
      onToggleActions={() => {}}
      focusType={FOCUS_POINT}
      focusArea={{
        x: 1,
        y: 1,
        width: 1,
        height: 1
      }}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
