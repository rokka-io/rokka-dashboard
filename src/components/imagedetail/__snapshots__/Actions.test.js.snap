// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions does render 1`] = `
<div
  className="rka-crop-actions is-active"
>
  <div
    className="rka-crop-action-container is-active"
  >
    <button
      className="rka-crop-action"
      id="focusPoint"
      onClick={[Function]}
    >
      <svg
        className="rka-focuspoint-icon"
      >
        <use
          xlinkHref="focuspoint-icon.svg#focuspointicon"
        />
      </svg>
    </button>
    <button
      className="rka-crop-action"
      id="focusArea"
      onClick={[Function]}
    >
      <svg
        className="rka-focusarea-icon"
      >
        <use
          xlinkHref="focusarea-icon.svg#focusareaicon"
        />
      </svg>
    </button>
  </div>
  <div
    className="rka-crop-actions-form is-active"
  >
    <div
      className="row"
    >
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Offset X
          </label>
          <input
            className="rka-input-txt"
            name="x"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Offset Y
          </label>
          <input
            className="rka-input-txt"
            name="y"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Width
          </label>
          <input
            className="rka-input-txt"
            disabled={false}
            name="width"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Height
          </label>
          <input
            className="rka-input-txt"
            disabled={false}
            name="height"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="rka-crop-actions-close-container"
  >
    <button
      className="rka-crop-actions-close"
      onClick={[Function]}
    >
      ×
    </button>
  </div>
</div>
`;

exports[`Actions does render with focus area 1`] = `
<div
  className="rka-crop-actions is-active"
>
  <div
    className="rka-crop-action-container is-active"
  >
    <button
      className="rka-crop-action"
      id="focusPoint"
      onClick={[Function]}
    >
      <svg
        className="rka-focuspoint-icon"
      >
        <use
          xlinkHref="focuspoint-icon.svg#focuspointicon"
        />
      </svg>
    </button>
    <button
      className="rka-crop-action is-active"
      id="focusArea"
      onClick={[Function]}
    >
      <svg
        className="rka-focusarea-icon"
      >
        <use
          xlinkHref="focusarea-icon.svg#focusareaicon"
        />
      </svg>
    </button>
  </div>
  <div
    className="rka-crop-actions-form is-active"
  >
    <div
      className="row"
    >
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Offset X
          </label>
          <input
            className="rka-input-txt"
            name="x"
            onChange={[Function]}
            type="text"
            value={1}
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Offset Y
          </label>
          <input
            className="rka-input-txt"
            name="y"
            onChange={[Function]}
            type="text"
            value={1}
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Width
          </label>
          <input
            className="rka-input-txt"
            disabled={false}
            name="width"
            onChange={[Function]}
            type="text"
            value={100}
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Height
          </label>
          <input
            className="rka-input-txt"
            disabled={false}
            name="height"
            onChange={[Function]}
            type="text"
            value={100}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="rka-crop-actions-close-container"
  >
    <button
      className="rka-crop-actions-close"
      onClick={[Function]}
    >
      ×
    </button>
  </div>
</div>
`;

exports[`Actions does render with focus point 1`] = `
<div
  className="rka-crop-actions is-active"
>
  <div
    className="rka-crop-action-container is-active"
  >
    <button
      className="rka-crop-action is-active"
      id="focusPoint"
      onClick={[Function]}
    >
      <svg
        className="rka-focuspoint-icon"
      >
        <use
          xlinkHref="focuspoint-icon.svg#focuspointicon"
        />
      </svg>
    </button>
    <button
      className="rka-crop-action"
      id="focusArea"
      onClick={[Function]}
    >
      <svg
        className="rka-focusarea-icon"
      >
        <use
          xlinkHref="focusarea-icon.svg#focusareaicon"
        />
      </svg>
    </button>
  </div>
  <div
    className="rka-crop-actions-form is-active"
  >
    <div
      className="row"
    >
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Offset X
          </label>
          <input
            className="rka-input-txt"
            name="x"
            onChange={[Function]}
            type="text"
            value={1}
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Offset Y
          </label>
          <input
            className="rka-input-txt"
            name="y"
            onChange={[Function]}
            type="text"
            value={1}
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Width
          </label>
          <input
            className="rka-input-txt"
            disabled={true}
            name="width"
            onChange={[Function]}
            type="text"
            value={1}
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Height
          </label>
          <input
            className="rka-input-txt"
            disabled={true}
            name="height"
            onChange={[Function]}
            type="text"
            value={1}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="rka-crop-actions-close-container"
  >
    <button
      className="rka-crop-actions-close"
      onClick={[Function]}
    >
      ×
    </button>
  </div>
</div>
`;

exports[`Actions does render with menu inactive 1`] = `
<div
  className="rka-crop-actions"
>
  <div
    className="rka-crop-action-container is-active"
  >
    <button
      className="rka-crop-action"
      id="focusPoint"
      onClick={[Function]}
    >
      <svg
        className="rka-focuspoint-icon"
      >
        <use
          xlinkHref="focuspoint-icon.svg#focuspointicon"
        />
      </svg>
    </button>
    <button
      className="rka-crop-action"
      id="focusArea"
      onClick={[Function]}
    >
      <svg
        className="rka-focusarea-icon"
      >
        <use
          xlinkHref="focusarea-icon.svg#focusareaicon"
        />
      </svg>
    </button>
  </div>
  <div
    className="rka-crop-actions-form is-active"
  >
    <div
      className="row"
    >
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Offset X
          </label>
          <input
            className="rka-input-txt"
            name="x"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Offset Y
          </label>
          <input
            className="rka-input-txt"
            name="y"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Width
          </label>
          <input
            className="rka-input-txt"
            disabled={false}
            name="width"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="col-md-3 col-sm-3"
      >
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Height
          </label>
          <input
            className="rka-input-txt"
            disabled={false}
            name="height"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="rka-crop-actions-close-container"
  >
    <button
      className="rka-crop-actions-close"
      onClick={[Function]}
    >
      ×
    </button>
  </div>
</div>
`;

exports[`Actions does render with menu/actions inactive 1`] = `
<div
  className="rka-crop-actions"
>
  <div
    className="rka-crop-action-container"
  >
    <button
      className="rka-crop-action"
      id="focusPoint"
      onClick={[Function]}
    >
      <svg
        className="rka-focuspoint-icon"
      >
        <use
          xlinkHref="focuspoint-icon.svg#focuspointicon"
        />
      </svg>
    </button>
    <button
      className="rka-crop-action"
      id="focusArea"
      onClick={[Function]}
    >
      <svg
        className="rka-focusarea-icon"
      >
        <use
          xlinkHref="focusarea-icon.svg#focusareaicon"
        />
      </svg>
    </button>
  </div>
  <div
    className="rka-crop-actions-form"
  />
</div>
`;
